src/targets/gpu/device/scatter.cpp

```diff
@@ -15,20 +15,22 @@ argument scatter(
 {
     auto ds            = arg0.get_shape();
     auto inds          = arg1.get_shape();
+    auto upds          = arg2.get_shape();
     auto axis_dim_size = ds.lens()[axis];
-    hip_visit_all(result, arg0, inds)([&](auto output, auto data, auto s1) {
+    hip_visit_all(result, arg0, inds, upds)([&](auto output, auto data, auto inds1, auto upds1) {
```

Comment:
I am not sure why we pass `inds` and `upds` shapes. We can get the shapes using `indices.get_shape()` and `update.get_shape()`.

Reply:
These changes are triggering a series of Compile-time errors.